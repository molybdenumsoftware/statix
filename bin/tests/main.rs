mod util {
    #[macro_export]
    macro_rules! test_lint {
        ($tname:ident, $($tail:tt)*) => {
            test_lint!($tname);
            test_lint!($($tail)*);
        };
        ($tname:ident, $($tail:tt)*) => {
                test_lint!($tname);
                test_lint!($($tail)*);
        };
        ($tname:ident) => {
            paste::paste! {
                #[test]
                fn [<$tname _lint>](){
                    let file_path = concat!("tests/data/", stringify!($tname), ".nix");
                    test_cli(concat!(stringify!($tname), "_lint"), file_path, &["check"]);
                }

                #[test]
                fn [<$tname _fix>](){
                    let file_path = concat!("tests/data/", stringify!($tname), ".nix");
                    test_cli(concat!(stringify!($tname), "_fix"), file_path, &["fix", "--dry-run"]);
                }
            }

        };
    }
}

fn test_cli(test_name: &str, file_path: &str, args: &[&str]) {
    let output = std::process::Command::new("cargo")
        .arg("run")
        .arg("--")
        .args(args)
        .arg(file_path)
        .output()
        .expect("command runs successfully");

    let stdout = strip_ansi_escapes::strip(output.stdout).unwrap();
    let stdout = String::from_utf8(stdout).unwrap();

    insta::assert_snapshot!(test_name, &stdout);
}

test_lint! {
    collapsible_let_in,
    eta_reduction,
    useless_parens,
    empty_pattern,
    redundant_pattern_bind,
    unquoted_uri,
    empty_inherit,
    deprecated_to_path,
    useless_has_attr,
    repeated_keys,
    empty_list_concat
}
