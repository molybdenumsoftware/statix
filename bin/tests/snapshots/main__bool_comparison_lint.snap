---
source: bin/tests/main.rs
expression: "& out"
---
[W01] Warning: Unnecessary comparison with boolean
   ╭─[data/bool_comparison.nix:3:4]
   │
 3 │   (a == true)
   ·    ────┬────  
   ·        ╰────── Comparing a with boolean literal true
───╯
[W01] Warning: Unnecessary comparison with boolean
   ╭─[data/bool_comparison.nix:4:4]
   │
 4 │   (b == true)
   ·    ────┬────  
   ·        ╰────── Comparing b with boolean literal true
───╯
[W01] Warning: Unnecessary comparison with boolean
   ╭─[data/bool_comparison.nix:5:4]
   │
 5 │   (true == c)
   ·    ────┬────  
   ·        ╰────── Comparing c with boolean literal true
───╯
[W01] Warning: Unnecessary comparison with boolean
   ╭─[data/bool_comparison.nix:6:4]
   │
 6 │   (true == d)
   ·    ────┬────  
   ·        ╰────── Comparing d with boolean literal true
───╯
[W01] Warning: Unnecessary comparison with boolean
   ╭─[data/bool_comparison.nix:9:4]
   │
 9 │   (e != true)
   ·    ────┬────  
   ·        ╰────── Comparing e with boolean literal true
───╯
[W01] Warning: Unnecessary comparison with boolean
    ╭─[data/bool_comparison.nix:10:4]
    │
 10 │   (f != false)
    ·    ─────┬────  
    ·         ╰────── Comparing f with boolean literal false
────╯
[W01] Warning: Unnecessary comparison with boolean
    ╭─[data/bool_comparison.nix:11:4]
    │
 11 │   (true != g)
    ·    ────┬────  
    ·        ╰────── Comparing g with boolean literal true
────╯
[W01] Warning: Unnecessary comparison with boolean
    ╭─[data/bool_comparison.nix:12:4]
    │
 12 │   (false != h)
    ·    ─────┬────  
    ·         ╰────── Comparing h with boolean literal false
────╯
[W01] Warning: Unnecessary comparison with boolean
    ╭─[data/bool_comparison.nix:19:4]
    │
 19 │   (false == false)
    ·    ───────┬──────  
    ·           ╰──────── Comparing false with boolean literal false
────╯
[W01] Warning: Unnecessary comparison with boolean
    ╭─[data/bool_comparison.nix:20:4]
    │
 20 │   (false == true)
    ·    ──────┬──────  
    ·          ╰──────── Comparing true with boolean literal false
────╯
[W01] Warning: Unnecessary comparison with boolean
    ╭─[data/bool_comparison.nix:21:4]
    │
 21 │   (true == false)
    ·    ──────┬──────  
    ·          ╰──────── Comparing false with boolean literal true
────╯
[W01] Warning: Unnecessary comparison with boolean
    ╭─[data/bool_comparison.nix:22:4]
    │
 22 │   (true == true)
    ·    ──────┬─────  
    ·          ╰─────── Comparing true with boolean literal true
────╯
[W01] Warning: Unnecessary comparison with boolean
    ╭─[data/bool_comparison.nix:25:4]
    │
 25 │   (false == m ? n)
    ·    ───────┬──────  
    ·           ╰──────── Comparing m ? n with boolean literal false
────╯
[W01] Warning: Unnecessary comparison with boolean
    ╭─[data/bool_comparison.nix:26:4]
    │
 26 │   (true == o ? p)
    ·    ──────┬──────  
    ·          ╰──────── Comparing o ? p with boolean literal true
────╯
